File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> nums_to_index_map;

        for (int i = 0; i < nums.size(); i++) {
            int compliment = target - nums[i];
            
            if (nums_to_index_map.count(compliment)) {
                return {nums_to_index_map[compliment], i};
            }

            nums_to_index_map[nums[i]] = i;
        }
        
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup={}
        for idx,num in enumerate(nums):
            complement = target-num
            if complement in lookup:
                return [lookup[complement],idx]
            lookup[num]=idx